src/features/dsComponents/body/BodyTable.tsx

```diff
@@ -1,226 +1,219 @@
-  { width: 98.03, component: <rect y='7' x='8.65' width="35" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 124, component: <rect y='7' x='8.65' width="55" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 109.19, component: <rect y='7' x='8.65' width="22" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 75.13, component: <rect y='7' x='8.65' width="26" height="14" rx="1" fill="#D5DADD"/> }
+  { width: 168, component: <rect y='1' x='8.65' width="6" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 185.2, component: <rect y='1' x='8.65' width="104" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 163.77, component: <rect y='1' x='8.65' width="17" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 113.7, component: <rect y='1' x='8.65' width="22" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 98.03, component: <rect y='1' x='8.65' width="35" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 124, component: <rect y='1' x='8.65' width="55" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 109.19, component: <rect y='1' x='8.65' width="22" height="15" rx="1" fill="#D5DADD"/> }
 ]
 
 const loadingHeaderColumns: Array<{ width: number, component: React.ReactElement }> = [
-  { width: 185.2, component: <rect y='13' x='8.62' width="85" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 163.77, component: <rect y='13' x='8.62' width="113" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 113.7, component: <rect y='13' x='8.62' width="61" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 98.03, component: <rect y='13' x='8.62' width="34" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 124, component: <rect y='13' x='8.62' width="42" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 109.19, component: <rect y='13' x='8.62' width="57" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 75.13, component: <rect y='13' x='8.62' width="43" height="14" rx="1" fill="#D5DADD"/> }
+  { width: 185.2, component: <rect y='1' x='8.62' width="85" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 163.77, component: <rect y='1' x='8.62' width="113" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 113.7, component: <rect y='1' x='8.62' width="61" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 98.03, component: <rect y='1' x='8.62' width="34" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 124, component: <rect y='1' x='8.62' width="42" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 109.19, component: <rect y='1' x='8.62' width="57" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 75.13, component: <rect y='1' x='8.62' width="43" height="15" rx="1" fill="#D5DADD"/> }
 ]
 
-export default class BodyTable extends React.Component<BodyTableProps> {
-  constructor (props: BodyTableProps) {
-    super(props)
-    this.state = {
-      bodyLength: 0
-    }
+const BodyTable: React.FC<BodyTableProps> = ({
+  rowCount,
+  headers = [],
+  qriRef,
+  loading = false
+}) => {
+  const [ list, setList ] = useState<Record<number, any>>({})
```

Comment:
because we are using local state here, it shouldn't be too difficult to refactor this to use the state tree

Rather than using `list` and `setList` from `useState`, we can pull list using a new`selectBody` selector (that would be defined in `src/features/dataset/state/datasetState.ts` or just add back the `body` props.

rather than using `setList` we should expect that the extended list will get selected from the reducer.